### PR TITLE
Disable checkbox for blocking migration issues

### PR DESCRIPTION
**Pull request by cssuh**
Disable database checkbox if there is a blocking issue (multiple log files, filestream, or db too large)
disable-checkbox

### CODE REVIEW

File: extensions/sql-migration/src/models/stateMachine.ts [hunk 1]
```diff
@@ -55,6 +55,12 @@ export enum NetworkContainerType {
 	NETWORK_SHARE
 }
 
+export enum BlockingIssues {
+	LOG_FILE = 'Azure SQL Managed Instance does not support multiple log files.',
```
Comment (alanrenmsft) [inline extensions/sql-migration/src/models/stateMachine.ts:59]: these strings should be localized and you don't need an enum type for it

File: extensions/sql-migration/src/dialog/assessmentResults/sqlDatabasesTree.ts [hunk 2]
```diff
@@ -802,15 +802,22 @@ export class SqlDatabaseTree {
 			// Reset the dbName list so that it is in sync with the table
 			this._dbNames = this._model._assessmentResults.databaseAssessments.map(da => da.name);
 			this._model._assessmentResults.databaseAssessments.forEach((db) => {
+				let selectable = true;
+				db.issues.forEach((issue) => {
```
Comment (alanrenmsft) [inline extensions/sql-migration/src/dialog/assessmentResults/sqlDatabasesTree.ts:806]: you should just use the array.find, which will return the first one that matches the criteria, you don't have to go through all the issues.

you can do it inline when you set the enabled property.

**Comment by aasimkhan30** [comment]
@cssuh are we sure that this is the best way to match the blocking issues. The rule text might easily change and break this feature. Matching based on rule IDs will be better and will make it easier to add/remove the rules.
What is check ID in assessmentsResultItem and can it be used to uniquely match these rules? Also, in a future PR it will be great if we can document assessment properties in this interface.

https://github.com/microsoft/azuredatastudio/blob/339d908d1d180a91f457aa80e6fa486c8d62a19c/extensions/mssql/src/mssql.d.ts#L523-L524

**Comment by cssuh** [comment]
@aasimkhan30 @alanrenmsft 
Yes I agree that a ruleID needs to be used, it is actually not passed from the Assessment Results so I am talking with the migration assessments team to have them add it, and will edit the PR once it is available

File: extensions/sql-migration/src/constants/strings.ts [hunk 3]
```diff
@@ -356,6 +356,9 @@ export const TARGET_PLATFORM = localize('sql.migration.target.platform', "Target
 export const WARNINGS_DETAILS = localize('sql.migration.warnings.details', "Warnings Details");
 export const ISSUES_DETAILS = localize('sql.migration.issues.details', "Issue Details");
 export const SELECT_DB_PROMPT = localize('sql.migration.select.prompt', "Click on SQL Server Instance or any of the databases on the left to view its details.");
+export const BLOCKING_ISSUE_LOG_FILES = localize('sql.migration.blocking.issue.log', "MultipleLogFiles");
```
Comment (aasimkhan30) [inline extensions/sql-migration/src/constants/strings.ts:359]: I don't think we need to localize these ids. Only the strings that are displayed need to be localized. Just make an array of blocking IDs and check if the id is in that array.

**Review by aasimkhan30 — CHANGES_REQUESTED** [review]

**Review by aasimkhan30 — APPROVED** [review]

**Pull request merged**